src/chat/ChatScreen.js

```diff
@@ -19,27 +23,86 @@ type Props = {
   },
 };
 
-export default class ChatScreen extends PureComponent<Props> {
+type State = {
+  editMessage: ?EditMessage,
+};
+
+class ChatScreen extends PureComponent<Props, State> {
   context: Context;
   props: Props;
+  state: State;
+
+  state = {
+    editMessage: null,
+  };
 
   static contextTypes = {
     styles: () => null,
   };
 
+  componentDidMount() {
+    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
```

Comment:
Not a great fan of all this handling.
I suspect this will not work with the back gesture on iOS.
That is one of the reasons I wanted to base this on React Navigation's route.

Reply:
Now, even if user uses iOS gesture then also he is out of `editMode` + navigateBack (totally what he expects) :)